closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -992,6 +982,33 @@ boolean isRestrictedProviderAllowed(String providerName) {
             return false;
         }
 
+        /**
+         * Get the provider name defined in provider construction method.
+         *
+         * @param providerName provider name or provider with packages or arguments
+         * @return provider name defined in provider construction method
+         */
+        private static String getProvidersSimpleName(String providerName) {
+            // Remove the provider's optional arguments if present.
+            int pos = providerName.indexOf(' ');
+            providerName = (pos < 0) ? providerName.trim() : providerName.substring(0, pos).trim();
```

Comment:
I'm not a fan of statements that can effectively assign a variable to itself. Please separate the `substring()` and `trim()` steps:
```java
            if (pos >= 0) {
                providerName = providerName.substring(0, pos);
            }
            providerName = providerName.trim();
```
Similarly line 998 should be:
```java
            if (pos >= 0) {
                providerName = providerName.substring(0, pos);
            }
```
and line 1006 should be:
```java
                if (pos >= 0) {
                    providerName = providerName.substring(0, pos + 1);
                }
```